### PR TITLE
[3.x] Mark debugger limits settings as requiring a restart

### PR DESCRIPTION
**Pull request by Calinou**
These settings are only read once on engine initialization.

- This closes https://github.com/godotengine/godot/issues/81694.

### CODE REVIEW

File: doc/classes/ProjectSettings.xml [hunk 1]
```diff
@@ -1242,7 +1242,7 @@
 			Default map up vector for 3D navigation maps. See [method NavigationServer.map_set_up].
 		</member>
 		<member name="network/limits/debugger_stdout/max_chars_per_second" type="int" setter="" getter="" default="2048">
-			Maximum amount of characters allowed to send as output from the debugger. Over this value, content is dropped. This helps not to stall the debugger connection.
+			Maximum amount of characters allowed to send as output from the debugger. Over this value, content is dropped with the message [code]"output overflow, print less text!"[/code]. This helps not to stall the debugger connection.
```
Comment (RandomShaper) [inline doc/classes/ProjectSettings.xml:1245]: Isn't it maybe better not to quote the message? No big deal, but I'm foreseeing the message being reworded at some point with the docs forgotten to be updated.

File: main/main.cpp [hunk 2]
```diff
@@ -997,15 +997,15 @@ Error Main::setup(const char *execpath, int argc, char *argv[], bool p_second_ph
 	// Initialize user data dir.
 	OS::get_singleton()->ensure_user_data_dir();
 
-	GLOBAL_DEF("memory/limits/multithreaded_server/rid_pool_prealloc", 60);
+	GLOBAL_DEF_RST("memory/limits/multithreaded_server/rid_pool_prealloc", 60);
```
Comment (RandomShaper) [inline main/main.cpp:1000]: If there are more settings that need the change, maybe this one could be put into another PR addressing a bunch of misc settings, given that this one doesn't belong to the category the PR title suggests.

**Review by RandomShaper — APPROVED** [review]
Regardless what you decide about my comments, this looks good to me.

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Oops I forgot to wait for Pedro's comments to be answered. Feel free to make a follow-up PR with changes if you want to make any.

**Comment by RandomShaper** [comment]
Oh, yeah, no problems. When I give an approval, I consider the code good enough as it is and the author can decide if follow my comments or not. Maybe not the best workfow, but I think is good for efficiency when I have lots or PRs to check, like a "fire and maybe forget."

**Comment by akien-mga** [comment]
Yeah I think it's a good workflow :+1: 
Usually I'd give such PR a few days for further comments but this time I was overeager in my 3.x merging spree :D

**Comment by akien-mga** [comment]
Cherry-picked for 3.5.3.
